plugins/plugin-urlfactory/src/main/java/org/eclipse/che/plugin/urlfactory/URLFetcher.java

```diff
@@ -40,6 +42,9 @@
   /** Maximum size of allowed data. (30KB) */
   protected static final long MAXIMUM_READ_BYTES = 30 * 1000;
 
+  /** The Compiled REGEX PATTERN that can be used for http|https git urls */
+  final Pattern GIT_HTTP_URL_PATTERN = Pattern.compile("(^http[s]?://.*)(.git$)");
```

Comment:
Please consider next regexp: `"(?<sanitized>^https?://.*)\\.git$"`
Then fetching sanitized part can be done in such a way: `matcher.group("sanitized")`

Reply:
FYI if one needs to match dot in regexp it should be escaped with 2 backslashes

Reply:
thanks @garagatyi  - thought of it and somehow slipped when i implemented it. thanks for it.. I have updated the code